src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -1326,13 +1287,13 @@ clean_pkgcache_orphans (RpmOstreeSysrootUpgrader *self,
   for (guint i = 0; i < deployments->len; i++)
     {
       OstreeDeployment *deployment = deployments->pdata[i];
-      GKeyFile *origin = ostree_deployment_get_origin (deployment);
-      gboolean is_local;
+      g_autoptr(RpmOstreeOrigin) origin = NULL;
 
-      if (!_rpmostree_origin_is_locally_assembled (origin, &is_local, error))
+      origin = rpmostree_origin_parse_deployment (deployment, error);
+      if (!origin)
         return FALSE;
 
-      if (is_local)
+      if (rpmostree_origin_is_locally_assembled (origin))
```

Comment:
Here too, we probably don't want to waste time investigating the deployment if no packages are layered but the initramfs is local.